internal/appsec/dyngo/instrumentation/httpsec/http.go

```diff
@@ -33,24 +33,32 @@ type (
 		Cookies map[string][]string
 		// Query corresponds to the address `server.request.query`
 		Query map[string][]string
+		// PathParams corresponds to the address `server.request.path_params`
+		PathParams map[string]string
 	}
 
 	// HandlerOperationRes is the HTTP handler operation results.
 	HandlerOperationRes struct {
 		// Status corresponds to the address `server.response.status`
 		Status int
 	}
+
+	// AppSecParams is the extra parameters that need to be passed over to the AppSec WrapHandler (i.e params
+	// not retrievable from the http.Request, such as path parameters)
+	AppSecParams struct {
+		PathParams map[string]string
+	}
 )
 
 // WrapHandler wraps the given HTTP handler with the abstract HTTP operation defined by HandlerOperationArgs and
 // HandlerOperationRes.
-func WrapHandler(handler http.Handler, span ddtrace.Span) http.Handler {
+func WrapHandler(handler http.Handler, span ddtrace.Span, params AppSecParams) http.Handler {
```

Comment:
I propose we simplify and further specialise this a bit with:
```suggestion
func WrapHandler(handler http.Handler, span ddtrace.Span, pathParams map[string]string) http.Handler {
```

Because the usual Go pattern for optional function arguments is to have a variadic function along with option values. But there's no need for that for now due to the fact we have this single option only today. So I would rather make it the "nullable argument of path parameters" for now.

This would simplify some other files you have.

WDYT?

Reply:
Agreed, I'll make the change